lib/core/bucketEntries/MongoDBBucketEntriesRepository.ts

```diff
@@ -18,58 +63,68 @@ export class MongoDBBucketEntriesRepository implements BucketEntriesRepository {
     if (where.id) {
       query = { ...query, _id: where.id };
       delete query.id;
-    } 
- 
-    const bucketEntry: BucketEntryModel | null = await this.model.findOne(query);
+    }
+
+    const bucketEntry: BucketEntryModel | null = await this.model.findOne(
+      query
+    );
 
-    return bucketEntry ? { id: bucketEntry._id, ...bucketEntry.toObject() } : bucketEntry;    
+    if (!bucketEntry) {
+      return null;
+    }
+
+    return formatFromMongoToBucketEntry(bucketEntry);
   }
 
   async findByIds(ids: string[]): Promise<BucketEntry[]> {
     const bucketEntries = await this.model.find({ _id: { $in: ids } });
 
-    return bucketEntries.map((b: any) => b.toObject());
+    return bucketEntries.map(formatFromMongoToBucketEntry);
   }
 
   async findOneWithFrame(
     where: Partial<BucketEntry>
-  ): Promise<Omit<BucketEntryWithFrame, 'frame'> & { frame?: Frame } | null> {
+  ): Promise<(Omit<BucketEntryWithFrame, 'frame'> & { frame?: Frame }) | null> {
     let query: Partial<BucketEntry> & { _id?: string } = where;
 
     if (where.id) {
       query = { ...query, _id: where.id };
       delete query.id;
-    } 
- 
-    const bucketEntry: Omit<BucketEntryModel, 'frame'> & { frame?: Frame } | null = await this.model
-      .findOne(query)
-      .populate('frame')
-      .exec();
+    }
+
+    const bucketEntry:
+      | (Omit<BucketEntryModel, 'frame'> & { frame?: Frame })
+      | null = await this.model.findOne(query).populate('frame').exec();
 
-    let result: Omit<BucketEntryWithFrame, 'frame'> & { frame?: Frame } | null = null;
+    let result:
+      | (Omit<BucketEntryWithFrame, 'frame'> & { frame?: Frame })
+      | null = null;
 
     if (bucketEntry) {
-      result = { 
-        ...bucketEntry.toObject(),
-        id: bucketEntry._id,
-        frame: bucketEntry.frame
-      };
+      return formatFromMongoToBucketEntryWithFrame(bucketEntry);
     }
 
-    return result;    
+    return result;
   }
 
-  async findByIdsWithFrames(ids: BucketEntry['id'][]): Promise<(Omit<BucketEntryWithFrame, "frame"> & { frame?: Frame | undefined; })[]> {
+  async findByIdsWithFrames(
+    ids: BucketEntry['id'][]
+  ): Promise<
+    (Omit<BucketEntryWithFrame, 'frame'> & { frame?: Frame | undefined })[]
```

Comment:
Using `?` and `undefined` is redundant, please, remove one of them

Reply:
fixed